ledger/stateproofverificationtracker.go

```diff
@@ -0,0 +1,244 @@
+	if err != nil {
+		return err
+	}
+
+	if dcc.stateProofVerificationLatestDeleteDataIndex > -1 {
+		err = deleteOldStateProofVerificationData(ctx, tx, dcc.stateProofVerificationDeleteData.stateProofNextRound)
+	}
+
+	return err
+
+}
+
+func (spt *stateProofVerificationTracker) postCommit(_ context.Context, dcc *deferredCommitContext) {
+	spt.stateProofVerificationMu.Lock()
+	defer spt.stateProofVerificationMu.Unlock()
+
+	spt.trackedCommitData = spt.trackedCommitData[len(dcc.stateProofVerificationCommitData):]
+	spt.trackedDeleteData = spt.trackedDeleteData[dcc.stateProofVerificationLatestDeleteDataIndex+1:]
+}
+
+func (spt *stateProofVerificationTracker) postCommitUnlocked(context.Context, *deferredCommitContext) {
+
+}
+
+func (spt *stateProofVerificationTracker) handleUnorderedCommit(*deferredCommitContext) {
+
+}
+
+func (spt *stateProofVerificationTracker) close() {
+	if spt.dbQueries.lookupStateProofVerificationData != nil {
+		spt.dbQueries.lookupStateProofVerificationData.Close()
+	}
+}
+
+func (spt *stateProofVerificationTracker) LookupVerificationData(stateProofLastAttestedRound basics.Round) (*ledgercore.StateProofVerificationData, error) {
+	spt.stateProofVerificationMu.RLock()
+	defer spt.stateProofVerificationMu.RUnlock()
+
+	if len(spt.trackedCommitData) == 0 || stateProofLastAttestedRound < spt.trackedCommitData[0].verificationData.TargetStateProofRound {
+		return spt.dbQueries.lookupData(stateProofLastAttestedRound)
+	}
+
+	if stateProofLastAttestedRound <= spt.trackedCommitData[len(spt.trackedCommitData)-1].verificationData.TargetStateProofRound {
+		verificationData, err := spt.lookupDataInTrackedMemory(stateProofLastAttestedRound)
+		return &verificationData, err
+	}
+
+	return &ledgercore.StateProofVerificationData{}, errStateProofVerificationDataNotYetGenerated
+}
+
+func (spt *stateProofVerificationTracker) committedRoundToLatestCommitDataIndex(committedRound basics.Round) int {
+	latestCommittedDataIndex := -1
+
+	for index, data := range spt.trackedCommitData {
+		if data.confirmedRound <= committedRound {
+			latestCommittedDataIndex = index
+		}
+	}
+
+	return latestCommittedDataIndex
+}
+
+func (spt *stateProofVerificationTracker) committedRoundToLatestDeleteDataIndex(committedRound basics.Round) int {
+	latestCommittedDataIndex := -1
+
+	for index, data := range spt.trackedDeleteData {
+		if data.confirmedRound <= committedRound {
+			latestCommittedDataIndex = index
+		}
+	}
+
+	return latestCommittedDataIndex
+}
+
+func (spt *stateProofVerificationTracker) lookupDataInTrackedMemory(stateProofLastAttestedRound basics.Round) (ledgercore.StateProofVerificationData, error) {
+	for _, commitData := range spt.trackedCommitData {
+		if commitData.verificationData.TargetStateProofRound == stateProofLastAttestedRound {
+			return commitData.verificationData, nil
+		}
+	}
+
+	return ledgercore.StateProofVerificationData{}, errStateProofVerificationDataNotFound
```

Comment:
I think we should use `errStateProofVerificationDataNotFound` instead (for example, it might be in the past)

Reply:
I'm not sure I follow. Instead of what?